lib/classes/Service.js

```diff
@@ -54,17 +54,13 @@ class Service {
     ];
 
     const serviceFilePaths = _.map(serviceFilenames, filename => path.join(servicePath, filename));
-    const serviceFileIndex = _.findIndex(serviceFilePaths,
+    const serviceFileIndex = Math.max(_.findIndex(serviceFilePaths,
       filename => this.serverless.utils.fileExistsSync(filename)
-    );
+    ), 0);
 
     // Set the filename if found, otherwise set the preferred variant.
-    const serviceFilePath = serviceFileIndex !== -1 ?
-      serviceFilePaths[serviceFileIndex] :
-      _.first(serviceFilePaths);
-    const serviceFilename = serviceFileIndex !== -1 ?
-      serviceFilenames[serviceFileIndex] :
-      _.first(serviceFilenames);
+    const serviceFilePath = serviceFilePaths[serviceFileIndex];
+    const serviceFilename = serviceFilenames[serviceFileIndex];
```

Comment:
These refactoring around `serviceFileIndex` is to increase test coverage so coverall does not complain.